test_log/src/main/java/org/corfudb/test_log/TestLog.java

```diff
@@ -0,0 +1,289 @@
+package org.corfudb.test_log;
+
+import org.apache.commons.codec.binary.Hex;
+import org.corfudb.format.Types.LogEntry;
+import org.corfudb.format.Types.LogHeader;
+import org.corfudb.format.Types.Metadata;
+import org.corfudb.format.Types.DataType;
+
+import java.nio.ByteBuffer;
+import java.lang.String;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.nio.channels.FileChannel;
+import java.io.IOException;
+import java.util.Map;
+
+import com.google.protobuf.ByteString;
+import com.google.common.hash.Hashing;
+import com.google.common.hash.HashFunction;
+import com.google.common.hash.HashCode;
+import org.docopt.Docopt;
+
+public class TestLog {
+    private static int MetadataSize;
+    private static int LogHeaderSize;
+    private static final String USAGE =
+            "Usage:\n"
+                    + "\ttest_log [-d <show binary> | -e <address> | -E <address>]"
+                    + "\n"
+                    + "Options:\n"
+                    + " -d <show binary>, --display=<show binary>   Display log records (include binary data)\n"
+                    + " -e <address>, --erase=<address>             Erase log record\n"
+                    + " -E <address>, --erase_remain=<address>      Erase log records to end of file\n";
```

Comment:
First, I would use a different letter than e here, to avoid a sticky-finger effect affecting a full erase.

Second, the full-name erase_remain sounds strange to me; perhaps use the full word 'erase_remaining' or 'erase_tail' or something like this.